artifacts/definitions/Linux/Remediation/Quarantine.yaml

```diff
@@ -0,0 +1,179 @@
+  hence the availability of nft CLI.
+
+  This artifact will create a table, with the default name
+  *vrr_quarantine_table*, which contains three chains. One
+  for inbound traffic, one for outbound traffic, and the other
+  for forwarding traffic. The chains will cut off all traffics
+  except those for DNS lookup and velociraptor itself.
+
+  To unquarantine the system, set the *RemovePolicy* parameter to *True*.
+
+precondition: SELECT OS From info() where OS = 'linux'
+
+type: CLIENT
+
+required_permissions:
+  - EXECVE
+
+parameters:
+  - name: pathToNFT
+    default: /usr/sbin/nft
+    description: We depend on nft to manage the tables, chains, and rules.
+
+  - name: TableName
+    default: vrr_quarantine_table
+    description: Name of the quarantine table
+
+  - name: MessageBox
+    description: |
+        Optional message box notification to send to logged in users. 256
+        character limit.
+
+  - name: RemovePolicy
+    type: bool
+    description: Tickbox to remove policy.
+
+sources:
+  - query: |
+     // If a MessageBox configured truncate to 256 character limit
+     LET MessageBox <= parse_string_with_regex(
+               regex='^(?P<Message>.{0,255}).*',
+               string=MessageBox).Message
+     // TODO(gyee): do we need to use the wall command or some equivalent to send the message?
+
+     // delete table
+     LET delete_table_cmd = (pathToNFT, 'delete', 'table', 'inet', TableName)
+
+     // add table
+     LET add_table_cmd = (pathToNFT, 'add', 'table', 'inet', TableName)
+
+     // add inbound chain
+     LET add_inbound_chain_cmd = (
+          pathToNFT, 'add', 'chain', 'inet', TableName, 'inbound_chain',
+          '{', 'type', 'filter', 'hook', 'input', 'priority', '0\;', 'policy', 'drop\;', '}')
+
+     // add tcp rule inbound chain to allow connections from Velociraptor
+     LET add_tcp_rule_to_inbound_chain_cmd = (
+          pathToNFT, 'add', 'rule', 'inet', TableName, 'inbound_chain',
+          'tcp', 'sport', '{', 'irdmi', '}', 'tcp', 'dport', '32768-65535',
+          'ct', 'state', 'established', 'accept')
+
+     // add udp rule inbound chain to allow DNS lookups
+     LET add_udp_rule_to_inbound_chain_cmd = (
+          pathToNFT,'add','rule','inet', TableName, 'inbound_chain',
+          'udp', 'sport', 'domain', 'udp', 'dport', '32768-65535', 
+          'ct', 'state', 'established', 'accept')
+
+     // add outbound chain
+     LET add_outbound_chain_cmd = (
+          pathToNFT, 'add', 'chain', 'inet', TableName, 'outbound_chain',
+          '{', 'type', 'filter', 'hook', 'output', 'priority', '0\;', 'policy', 'drop\;', '}')
+
+     // add tcp rule outbound chain to allow connections to Velociraptor
+     LET add_tcp_rule_to_outbound_chain_cmd = (
+          pathToNFT, 'add', 'rule', 'inet', TableName, 'outbound_chain',
+          'tcp', 'dport', '{', 'irdmi', '}', 'tcp', 'sport', '32768-65535',
+          'ct', 'state', 'new,established', 'accept')
```

Comment:
This should be tied to the server configuration. The Windows.Remediation.Quarantine artifact has code that can be reused for this.

Reply:
Good point! Thanks for the tip.